tests/unit/mathematics/linalg/Linalg_dot_unittest.cc

```diff
@@ -0,0 +1,60 @@
+#include <shogun/lib/config.h>
+
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+
+#ifdef HAVE_VIENNACL
+#include <shogun/mathematics/linalg/LinalgBackendViennaCL.h>
+#endif
+
+#include <shogun/lib/SGVector.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+TEST(LinalgDot, linalg_cpu_backend_dot)
```

Comment:
Also LinalgDot seems not the correct name here, we always use the class name
Here that would be `LinalgBackendEigen` no?